src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/BinderHelper.cs

```diff
@@ -15,22 +15,25 @@ namespace Microsoft.CSharp.RuntimeBinder
 {
     internal static class BinderHelper
     {
+        private static MethodInfo s_DoubleIsNaN;
+        private static MethodInfo s_SingleIsNaN;
+
         internal static DynamicMetaObject Bind(
                 DynamicMetaObjectBinder action,
                 RuntimeBinder binder,
-                IEnumerable<DynamicMetaObject> args,
+                DynamicMetaObject[] args,
                 IEnumerable<CSharpArgumentInfo> arginfos,
```

Comment:
Potentially this could be made an `IList<CSharpArgumentInfo>` or perhaps even a `CSharpArgumentInfo[]` but I think I've found a flaw that needs to be fixed first, so I'm investigating that.